src/test/java/com/wikia/webdriver/testcases/onsitenotifications/OnSiteNotificationsTests.java

```diff
@@ -0,0 +1,206 @@
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = MOBILE)
+  public void userOnMobileReceivesPostUpvoteNotification() {
+    Notification notification = createPostUpvoteNotification(User.USER_12, User.USER_2);
+
+    Assertion.assertTrue(getNotificationsMobile().contains(notification));
+  }
+
+  @Execute(asUser = User.USER_12, onWikia = WIKI_MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = MOBILE)
+  public void userOnMobileReceivesReplyUpvoteNotification() {
+    Notification notification = createReplyUpvoteNotification(User.USER_12, User.USER_12, User.USER_2);
+
+    Assertion.assertTrue(getNotificationsMobile().contains(notification));
+  }
+
+  @Execute(asUser = User.USER_12, onWikia = WIKI_MOBILE)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  @Test(groups = MOBILE)
+  public void userOnMobileSeesConsolidatedReplyNotification() {
+    Notification notification = createConsolidatedPostReplyNotification(User.USER_12);
+    Notifications notificationsList = getNotificationsMobile();
+
+    Assertion.assertTrue(notificationsList.contains(notification));
+  }
+
+  /**
+   * Helper methods
+   */
+
+  private PostEntity.Data createPostAs(User user) {
+    return DiscussionsOperations.using(user, driver).createPostWithUniqueData(siteId);
+  }
+
+  private ReplyEntity.Data createReplyToPostAs(PostEntity.Data post, User user) {
+    return DiscussionsOperations.using(user, driver).createReplyToPost(siteId, post);
+  }
+
+  private void upvotePostAs(PostEntity.Data post, User user) {
+    DiscussionsOperations.using(user, driver).upvotePost(siteId, post);
+  }
+
+  private void upvoteReplyAs(ReplyEntity.Data reply, User user) {
+    DiscussionsOperations.using(user, driver).upvoteReply(siteId, reply);
+  }
+
+  private Notification createPostReplyNotification(User postAuthor, User replyAuthor) {
+    PostEntity.Data post = createPostAs(postAuthor);
+    createReplyToPostAs(post, replyAuthor);
+    return NotificationFactory.getPostReplyNotification(replyAuthor, post);
+  }
+
+  private Notification createPostUpvoteNotification(User postAuthor, User upvoteAuthor) {
+    PostEntity.Data post = createPostAs(postAuthor);
+    upvotePostAs(post, upvoteAuthor);
+    return NotificationFactory.getPostUpvoteNotification(post);
+  }
+
+  private Notification createReplyUpvoteNotification(User postAuthor, User replyAuthor, User upvoteAuthor) {
+    PostEntity.Data post = createPostAs(postAuthor);
+    ReplyEntity.Data reply = createReplyToPostAs(post, replyAuthor);
+    upvoteReplyAs(reply, upvoteAuthor);
+    return NotificationFactory.getReplyUpvoteNotification();
+  }
+
+  private Notification createConsolidatedPostReplyNotification(User postAuthor) {
+    PostEntity.Data post = createPostAs(postAuthor);
+    for(User replyAuthor : replyUsers) {
+      createReplyToPostAs(post, replyAuthor);
+    }
```

Comment:
Use the sexy `replyUsers.forEach(replyAuthor -> createReplyToPostAs(post, replyAuthor);`!